File: OldSnacks/Java/CreditCardValidator/CreditCard.java
import java.util.Date;
import java.time.LocalDate; 
import java.util.stream.IntStream;
import java.util.stream.Collectors;
public class CreditCard{
	public static boolean validateCardLength(int[] array){
		if(array.length >= 13 && array.length <= 16){
			return true;
		}
		return false;
	}
	public static String checkCardType(int[] array){
		if(array.length >= 13 && array.length <= 16){
			if(array[0] == 4){
				return "Visa Card";
			}
			else if(array[0] == 5){
				return "MasterCard";
			}
			else if(array[0] == 6){
				return "Discover Card";
			}
			else if(array[0] == 3 && array[1] == 7){
				return "American Express Card";
			}else{
				return "Invalid Card";
			}
		}
		return "Invalid Card";
	}	
	public static int getCardLength(int[] array){
		return array.length;
	}
	public static String printCardNumbers(int[] array){
		String message = IntStream.of(array)
         		 		  .mapToObj(num -> String.valueOf(num))
					  .collect(Collectors.joining());
		return message;
	}
	public static boolean checkForNegative(int[] array){
		for(int counter = 0; counter < array.length; counter++){
			if(array[counter] < 0 || array[counter] > 9){
				return true;
			}
		}
		return false;
	}
	public static int firstStep(int[] array){
		int result = 0;
		int sum = 0;
		for(int counter = 0; counter < array.length; counter++){
			if(counter % 2 == 0){
				result = array[counter] * 2;
				if(result > 9){
					result -= 9;
				}
				sum += result;
			}
			 
		}
		return sum;
	}
	public static int addDigitsInOdd(int[] array){
		int total = 0;
		for(int counter = 0; counter < array.length; counter++){
			if(counter % 2 != 0){
				total += array[counter];
			}
		}
		return total;
	}
	public static int getSum(int[] array){
		int evenTotal = firstStep(array);
		int oddTotal =  addDigitsInOdd(array);
		return evenTotal + oddTotal;
	}
	public static boolean checkCard(int[] array){
		int total = getSum(array);
		if(total % 10 == 0){
			return true;
		}
		return false;
	}
	public static String showStatus(boolean status){
		if(status){
			return "Valid";
		}
		return "Invalid";
	} 
	public static String printOutput(int[] array){
		String cardType = checkCardType(array);
		String cardNumber = printCardNumbers(array);
		int cardLength = getCardLength(array);
		String validity = showStatus(checkCard(array));
		LocalDate time = LocalDate.now();
		Date now = new Date();
		String message = """


**********************************************
**Credit Card Type: %s
**Credit Card Number: %s
**Credit Card Length: %d
**Credit Card Validity Status: %s

		Date: %s %tT
**********************************************
		""".formatted(cardType, cardNumber, cardLength, validity, time, now);
		return message;
	}
}

File: OldSnacks/Python/CreditCard/credit_card.py
from datetime import datetime, date

def validate_card_length(array):
    return 13 <= len(array) <= 16

def check_card_type(array):
    if 13 <= len(array) <= 16:
        if array[0] == 4:
            return "Visa Card"
        elif array[0] == 5:
            return "MasterCard"
        elif array[0] == 6:
            return "Discover Card"
        elif array[0] == 3 and len(array) > 1 and array[1] == 7:
            return "American Express Card"
        else:
            return "Invalid Card"
    return "Invalid Card"

def get_card_length(array):
    return len(array)

def print_card_numbers(array):
    return ''.join(str(num) for num in array)

def check_for_negative(array):
    for num in array:
        if num < 0 or num > 9:
            return True
    return False

def first_step(array):
    result = 0
    total = 0
    for i in range(len(array)):
        if i % 2 == 0:
            result = array[i] * 2
            if result > 9:
                result -= 9
            total += result
    return total

def add_digits_in_odd(array):
    total = 0
    for i in range(len(array)):
        if i % 2 != 0:
            total += array[i]
    return total

def get_sum(array):
    even_total = first_step(array)
    odd_total = add_digits_in_odd(array)
    return even_total + odd_total

def check_card(array):
    total = get_sum(array)
    return total % 10 == 0

def show_status(status):
    return "Valid" if status else "Invalid"

def print_output(array):
    card_type = check_card_type(array)
    card_number = print_card_numbers(array)
    card_length = get_card_length(array)
    validity = show_status(check_card(array))
    today = date.today()
    now = datetime.now()

    message = f"""


**********************************************
**Credit Card Type: {card_type}
**Credit Card Number: {card_number}
**Credit Card Length: {card_length}
**Credit Card Validity Status: {validity}

        Date: {today} {now.strftime('%H:%M:%S')}
**********************************************
    """
    return message

def convert_string_to_array(str_numbers):
    numbers = []
    for ch in str_numbers:
        number = int(ch)
        numbers.append(number)
    return numbers

